lib/ansible/modules/cloud/amazon/ec2_spot_pricing_history_info.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/python
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = """
+---
+module: ec2_spot_pricing_history_facts
+short_description: Gather facts about ec2 spot pricing history in AWS
+description:
+  - Gather historical facts on the spot prices of different instance types
+version_added: "2.8"
+requirements: [ boto3>=1.9.0 ]
+author: "Dani Hodovic (@danihodovic)"
+options:
+  availability_zone:
+    description:
+      - >
+        availability zone for which prices should be returned. Returns prices
+        for all availability zones by default.
+    default: ""
+
+  instance_types:
+    description:
+      - the types of instances (for example, m3.medium),
+        see U(https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-types.html).
+        Returns prices for all instance types by default.
+    default: ""
+
+  product_descriptions:
+    description:
+      - the product description for the Spot price. Returns prices for all
+        product descriptions by default.
+    choices:
+      - 'Linux/UNIX'
+      - 'SUSE Linux'
+      - 'Windows'
+      - 'Linux/UNIX (Amazon VPC)'
+      - 'SUSE Linux (Amazon VPC)'
+      - 'Windows (Amazon VPC)'
+    default: []
+
+  start_time:
+    description:
+      - >
+        the date and time, up to the past 90 days, from which to start
+        retrieving the price history data. Defaults to the time right now.
+    default: "now"
+
+  end_time:
+    description:
+      - >
+        the date and time, up to the current date, from which to stop retrieving
+        the price history data. Defaults to the time right now.
+    default: "now"
+
+  max_results:
+    description:
+      - the total number of items to return in the output.
+    default: 1000
+
+  filters:
+    description:
+      - One or more filters
+    default: []
+
+extends_documentation_fragment:
+    - aws
+    - ec2
+"""
+
+EXAMPLES = """
+# Find the most recent spot price for t3.medium in eu-west-1a.
+# Facts are published as `ec2_spot_pricing_history`
+- ec2_spot_pricing_history_facts:
+    instance_types: ["t3.medium"]
+    availability_zone: "eu-west-1a"
+    product_descriptions: ["Linux/UNIX"]
+    max_results: 1
+  register: spot_prices
+
+# Create an instance using the spot price
+- ec2:
+    spot_price: "{{ ec2_spot_pricing_history[0].spot_price }}"
```

Comment:
Should match the variable registered in the preceding task.
```suggestion
    spot_price: "{{ spot_prices.ec2_spot_pricing_history.0.spot_price }}"
```